datafusion/physical-plan/src/sorts/sort_filters.rs

```diff
@@ -0,0 +1,297 @@
+                        replace = false;
+                        break;
+                    }
+                    if new_value_is_null && !current_value_is_null && !nulls_first {
+                        replace = false;
+                        break;
+                    }
+                    // Handle the descending case
+                    if descending {
+                        if new_value_is_greater_than_current {
+                            replace = false;
+                            break;
+                        }
+                    } else if !new_value_is_greater_than_current {
+                        replace = false;
+                        break;
+                    }
+                    // Handle the equality case
+                    if new_value.eq(current_value) {
+                        replace = false;
+                        break;
+                    }
+                }
+            }
+            replace
+        };
+        if replace {
+            let mut thresholds = self.thresholds.write().map_err(|_| {
+                datafusion_common::DataFusionError::Execution(
+                    "Failed to acquire write lock on thresholds".to_string(),
+                )
+            })?;
+            for (i, new_value) in new_values.iter().enumerate() {
+                thresholds[i] = Some(new_value.clone());
+            }
+        }
+        Ok(())
+    }
+
+    pub fn as_physical_expr(self: &Arc<Self>) -> Result<Arc<dyn PhysicalExpr>> {
+        let children = self
+            .expr
+            .iter()
+            .map(|sort_expr| Arc::clone(&sort_expr.expr))
+            .collect::<Vec<_>>();
+        Ok(Arc::new(DynamicFilterPhysicalExpr::new(
+            children,
+            Arc::clone(self) as Arc<dyn DynamicFilterSource>,
+        )))
+    }
+}
+
+impl Display for SortDynamicFilterSource {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let thresholds = self
+            .snapshot_current_filters()
+            .map_err(|_| std::fmt::Error)?
+            .iter()
+            .map(|p| format!("{p}"))
+            .collect::<Vec<String>>();
+        let inner = thresholds.join(",");
+        write!(f, "SortDynamicFilterSource[ {} ]", inner,)
+    }
+}
+
+impl DynamicFilterSource for SortDynamicFilterSource {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn snapshot_current_filters(&self) -> Result<Vec<Arc<dyn PhysicalExpr>>> {
+        let thresholds = self.thresholds.read().map_err(|_| {
+            datafusion_common::DataFusionError::Execution(
+                "Failed to acquire read lock on thresholds".to_string(),
+            )
+        })?;
+        // Create filter expressions for each threshold
+        let mut filters: Vec<Arc<dyn PhysicalExpr>> =
+            Vec::with_capacity(thresholds.len());
+
+        let mut prev_sort_expr: Option<Arc<dyn PhysicalExpr>> = None;
```

Comment:
I think this can be written with fewer expressions, like the following:

`col0 < threshold0 || (col0  == threshold0 && col1 < threshold1 || (...) `